pkg/resource/claim_reconciler.go

```diff
@@ -370,6 +372,21 @@ func (r *ClaimReconciler) Reconcile(req reconcile.Request) (reconcile.Result, er
 			return reconcile.Result{RequeueAfter: aShortWait}, errors.Wrap(r.client.Status().Update(ctx, claim), errUpdateClaimStatus)
 		}
 
+		if !IsConditionTrue(claim.GetCondition(v1alpha1.TypeReferencesResolved)) {
+			if err := r.managed.ResolveReferences(ctx, class); err != nil {
+				condition := v1alpha1.ReconcileError(err)
+				if IsReferencesAccessError(err) {
+					condition = v1alpha1.ReferenceResolutionBlocked(err)
+				}
+
+				claim.SetConditions(condition)
+				return reconcile.Result{RequeueAfter: aShortWait}, errors.Wrap(r.client.Status().Update(ctx, claim), errUpdateClaimStatus)
+			}
+
+			// Add ReferenceResolutionSuccess to claim conditions
+			claim.SetConditions(v1alpha1.ReferenceResolutionSuccess())
+		}
+
```

Comment:
Per discussion this morning I'd like to avoid resolving resource class references if possible. I believe it should be unnecessary, as the managed resource reconciler will take over once the claim reconciler has dynamically provisioned a managed resource.